### ISSUE TITLE
What is the best way given a TomlValue to get the string representation of it?

### ISSUE THREAD

**Issue by AndrewSav** [paiden/Nett#63]
I tired .Get but it's throwing different errors depending on the combination of value type and generic type. 

Examples:

```
Console.WriteLine(Toml.ReadString("a=\"b\"")["a"].Get<object>());
```

Produces:

```
FormatException: Guid should contain 32 digits with 4 dashes (xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx).
```

And:

```
Console.WriteLine(Toml.ReadString("a=1979-05-27T07:32:00-08:00")["a"].Get<string>());
```

Produces:

```
InvalidOperationException: Cannot convert from type 'date time' to 'String'.
```

**Comment by paiden** [comment]
Your first example should be:
```chsarp
Console.WriteLine(Toml.ReadString("a=\"b\"")["a"].Get<string>());
```

I admit the behavior when using no conversion at all - as the type arg is object - is not very good. I'll check if I can improve that behavior in the future.

The TomlDateTimeOffset can not be converted to a string directly. This is by design, as I'm not a fan of implicit conversions as I was bitten by too many subtle bugs because of them in the past.


Your second example will work with the following modification:


```csharp
Console.WriteLine(Toml.ReadString("a=1979-05-27T07:32:00-08:00")["a"].Get<DateTimeOffset>().ToString());

```

**Comment by AndrewSav** [comment]
@paiden Thank you for this, it does not answer the question though. I need a uniform way to print a string representation of a TomlValue. You gave to different ways, one of which won't work for dates and the other won't work for strings, which does not solve the problem.

Of course I can write a giant switch statement and write different conversions for each possible type, but that's usually what libraries are for. So... impossible to do it easily and naturally?

**Comment by AndrewSav** [comment]
An easy way to get what I want would be this:

```
	Console.WriteLine(((dynamic)Toml.ReadString("a=\"b\"")["a"]).Value);
	Console.WriteLine(((dynamic)Toml.ReadString("a=1979-05-27T07:32:00-08:00")["a"]).Value);
```

Unfortunately this incurs a heavy run-time penalty because of use of `dynamic`. Run-time dynamic dispatch is notorious for reduced performance. It has a distinct advantage of being simple/succinct though.

**Comment by paiden** [comment]
Fixed in 0.10.1 release
